parsers/DO.py

```diff
@@ -0,0 +1,285 @@
+    Sums type instances and returns a dictionary.    
+    """
+
+    #We only need the current hour.
+    lastt = df.tail(1)
+
+    #Create current plant output.
+    tp = {}
+    for item in list(df):
+        v = lastt.iloc[0][item]
+        tp[item] = v
+
+    current_plants = {k: tp[k] for k in tp if not isnan(tp[k])}
+    mapped_plants = [(thermal_plants[plant], val) for plant, val in current_plants.iteritems()]
+
+    thermalDict = defaultdict(lambda: 0.0)
+
+    #Sum values for duplicate keys.
+    for key,val in mapped_plants:
+        thermalDict[key] += val
+
+    return thermalDict
+
+
+def total_production(df):
+    """
+    Takes DataFrame and finds generation totals for current hour.
+    Returns a dictionary.
+    """
+
+    #The Dominican Republic does not observe daylight savings time.
+    find_hour = df.index[-1] 
+    at = arrow.now('UTC-4')
+    dt = (at.replace(hour = int(find_hour), minute = 0, second = 0)).datetime
+
+    latest = df.tail(1)
+    hydro = latest.iloc[0]['Hydro']
+    wind = latest.iloc[0]['Wind']
+
+    #Wind and hydro totals do not always update exactly on the new hour.
+    #In this case we set them to None because they are unknown rather than zero.
+    if isnan(wind):
+        wind = None
+    if isnan(hydro):
+        hydro = None
+    
+    prod = {'wind': wind, 'hydro': hydro, 'datetime': dt}
+
+    return prod
+
+
+def fetch_production(country_code = 'DO', session = None):
+    """
+    Requests the last known production mix (in MW) of a given country
+    Arguments:
+    country_code (optional) -- used in case a parser is able to fetch multiple countries
+    Return:
+    A dictionary in the form:
+    {
+      'countryCode': 'FR',
+      'datetime': '2017-01-01T00:00:00Z',
+      'production': {
+          'biomass': 0.0,
+          'coal': 0.0,
+          'gas': 0.0,
+          'hydro': 0.0,
+          'nuclear': null,
+          'oil': 0.0,
+          'solar': 0.0,
+          'wind': 0.0,
+          'geothermal': 0.0,
+          'unknown': 0.0
+      },
+      'storage': {
+          'hydro': -10.0,
+      },
+      'source': 'mysource.com'
+    }
+    """
+
+    dat = data_formatter(get_data(session = None))
+    tot = data_parser(dat['totals'])
+    th = data_parser(dat['thermal'])
+    thermal = thermal_production(th)
+    total = total_production(tot)
+
+    production_mix = {
+      'countryCode': country_code,
+      'datetime': total['datetime'],
+      'production': {
+          'biomass': thermal['biomass'],
+          'coal': thermal['coal'],
+          'gas': thermal['gas'],
+          'hydro': total['hydro'],
+          'nuclear': 0.0,
+          'oil': thermal['oil'],
+          'solar': None,
+          'wind': total['wind'],
+          'geothermal': 0.0,
+          'unknown': thermal['unknown']
+      },
+      'storage': {
+          'hydro': None,
+      },
+      'source': 'oc.org.do'
+    }
+    
+    return production_mix
```

Comment:
Just to be clear you want all the data available on the page?  So fetch_production() should return a list of 1-24 dicts (depends on the hour) each in the same format as production mix?